// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

import "google/protobuf/timestamp.proto";

package SafeEntry;



// The greeting service definition.
service SafeEntry {
  rpc Message (Request) returns (Reply) {}
  rpc Check (StatusCheckRequest) returns (StatusCheckResponse) {}
}

service LocationData {
    rpc GetLocation (get_location_data) returns (location) {}
}

// The request message containing the request
message Request {
  string message = 1;
}

// The response message containing the reply
message Reply {
  string message = 1;
}

message StatusCheckRequest {
  string status = 1;
}

message StatusCheckResponse {
  enum SafeEntryStatus {
    CHECKIN = 0;
    CHECKOUT = 1;
  }
  SafeEntryStatus status = 1;
}

// Location Data
//request for location data
message get_location_data {
  string location = 1;
}

//location value pair
message LocationInfo {
    string ic = 1;
    string name = 2;
    string checkin = 3;
    string checkout = 4;
    string infected = 5;
  }

// response containing the data
message location {
    repeated LocationInfo response = 1;
}

message UserInfo {
    string nric = 1;
    string name = 2;
    string location = 3;
}

message UserInfoDict {
  repeated UserInfo info = 1;
}
